api/spec/src/billing/invoices/invoice.tsp

```diff
@@ -0,0 +1,361 @@
+   * - draft, gathering: The time the invoice will be issued based on the workflow settings.
+   * - issued: The time the invoice was issued.
+   */
+  @summary("The time the invoice was issued.")
+  @visibility("read", "query")
+  issuedAt?: DateTime;
+
+  @summary("The workflow settings associated with this invoice")
+  @visibility("read", "query", "update")
+  workflow?: InvoiceWorkflowSettings;
+}
+
+/* Omitted fields (not required yet):
+ * op_date: Date when the operation defined by the invoice became effective.
+ * value_date: When the taxes of this invoice become accountable, if none set, the issue date is used.
+ * exchange_rates: Exchange rates to be used when converting the invoices monetary values into other currencies.
+ *    -> We are not supporting multi-currency invoices yet, so this is not supported.
+ * charges: 	Charges or surcharges applied to the complete invoice.
+ *    -> We are not supporting this yet.
+ * outlays: 	Expenses paid for by the supplier but invoiced directly to the customer.
+ *    -> We are not supporting this yet.
+ * ordering: Ordering details including document references and buyer or seller parties.
+ *    -> We are not supporting this yet (e.g. order handled by seperate entity, physical goods delivery details, etc.)
+ * delivery: Specific details on delivery of the goods referenced in the invoice.
+ *    -> We are not supporting physical goods delivery yet.
+ * tax: Special tax configuration for billing.
+ *    -> The prices include is the meaningful part, but we should make that an array, let's see if we need this as we implement more features.
+ */
+@summary("Invoice represents a document that is sent to a customer to request payment for goods or services.")
+@friendlyName("BillingInvoice")
+model Invoice {
+  ...InvoiceSummary;
+
+  @summary("List of invoice lines representing each of the items sold to the customer.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  lines?: Line[];
+
+  @summary("Discounts or allowances applied to the complete invoice.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  discounts?: Discount[];
+
+  @summary("Information on when, how, and to whom the invoice should be paid.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  payment?: Payment;
+}
+
+@friendlyName("BillingInvoiceWorkflowSettings")
+model InvoiceWorkflowSettings {
+  /**
+   * updatedAt is the time when the invoice workflow was last updated.
+   */
+  @summary("Should the invoice auto advance to the next status?")
+  autoAdvance: boolean;
+
+  /**
+   * The invoice stays in the draft status until this time.
+   *
+   * If auto advance is disabled the value will be empty.
+   */
+  draftUtil?: DateTime;
+
+  /**
+   * The period after which the invoice is due.
+   *
+   * default: 7 days
+   */
+  @encode(DurationKnownEncoding.ISO8601)
+  @example(duration.fromISO("P1D"))
+  dueAfter?: duration;
```

Comment:
use TS default as https://typespec.io/docs/language-basics/models#default-values

Reply:
@tothandras Duration defaults doesn't seem to work:

```
  itemCollectionPeriod?: duration = duration.fromISO("PT1H");
```

```
<unknown location>:1:1 - error @typespec/openapi3/invalid-default: Invalid type 'ScalarValue' for a default value
<unknown location>:1:1 - error @typespec/openapi3/invalid-default: Invalid type 'ScalarValue' for a default value
<unknown location>:1:1 - error @typespec/openapi3/invalid-default: Invalid type 'ScalarValue' for a default value
<unknown location>:1:1 - error @typespec/openapi3/invalid-default: Invalid type 'ScalarValue' for a default value
```

With that said I have commented out the other parts.

Reply:
Conclusion is that we use string type, but period encoder.